docs/sources/getting-started/ReactiveUI-101.md

```diff
@@ -26,90 +29,154 @@ public MainWindow()
-                .Where(x => !String.IsNullOrWhiteSpace(x))
-                .InvokeCommand(ExecuteSearch);
-
-            _SpinnerVisibility = ExecuteSearch.IsExecuting.Select(x=> x? Visibility.Visible : Visibility.Collapsed)                
-                .ToProperty(this, x => x.SpinnerVisibility, Visibility.Hidden);
-
-
-            IObservable<List<FlickrPhoto>> results;
-            if (testSearchResults != null)
-            {
-                results = testSearchResults;
-            }
-            else
-            {
-                results = ExecuteSearch.Select(term => GetSearchResultsFromFlickr((string)term));
-            }
-
-            _SearchResults = results.ToProperty(this, x => x.SearchResults, new List<FlickrPhoto>());
-        }
+    /* ObservableAsPropertyHelper
+     * 
+     * Here's the interesting part: In ReactiveUI, we can take IObservables
+     * and "pipe" them to a Property - whenever the Observable yields a new
+     * value, we will notify ReactiveObject that the property has changed.
+     * 
+     * To do this, we have a class called ObservableAsPropertyHelper - this
+     * class subscribes to an Observable and stores a copy of the latest value.
+     * It also runs an action whenever the property changes, usually calling
+     * ReactiveObject's RaisePropertyChanged.
+     */
 
-        public static List<FlickrPhoto> GetSearchResultsFromFlickr(string searchTerm)
-        {
-            var doc = XDocument.Load(String.Format(CultureInfo.InvariantCulture,
-                "http://api.flickr.com/services/feeds/photos_public.gne?tags={0}&format=rss_200",
-                HttpUtility.UrlEncode(searchTerm)));
+    ObservableAsPropertyHelper<List<FlickrPhoto>> _SearchResults;
+    public List<FlickrPhoto> SearchResults
+    {
+        get { return _SearchResults.Value; }
+    }
 
-            if (doc.Root == null)
-                return null;
+    // Here, we want to create a property to represent when the application 
+    // is performing a search (i.e. when to show the "spinner" control that 
+    // lets the user know that the app is busy). We also declare this property
+    // to be the result of an Observable (i.e. its value is derived from 
+    // some other property)
 
-            var titles = doc.Root.Descendants("{http://search.yahoo.com/mrss/}title")
-                .Select(x => x.Value);
+    ObservableAsPropertyHelper<Visibility> _SpinnerVisibility;
+    public Visibility SpinnerVisibility
+    {
+        get { return _SpinnerVisibility.Value; }
+    }
 
-            var tagRegex = new Regex("<[^>]+>", RegexOptions.IgnoreCase);
-            var descriptions = doc.Root.Descendants("{http://search.yahoo.com/mrss/}description")
-                .Select(x => tagRegex.Replace(HttpUtility.HtmlDecode(x.Value), ""));
+    public AppViewModel()
+    {
+        ExecuteSearch = ReactiveCommand.CreateAsyncTask(parameter => GetSearchResultsFromFlickr(this.SearchTerm));
```

Comment:
I wouldn't worry too much about cramming in *every* concept here. I mostly wanted `ThrownExceptions` here so that the user is aware about how to handle errors (rather than doing it inside `GetSearchResultsFromFlickr`)